### PR TITLE
Helpers to generate config.yml sections

### PR DESCRIPTION
**Pull request by pleasemarkdarkly**
This is what I used to speed up my population of the config.yml file and perhaps help someone trying this dotfiles setup.

### CODE REVIEW

File: gen-config-helper.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
+# config.yml helper
+In adapting [mac-dev-playbook](https://github.com/geerlingguy/mac-dev-playbook), preparing the config.yml file can be somewhat tedious. I found the following bash one-liners helpful, you may too. You'll want to clean up the output, but it should get you most of the way there.
+
+### Brewfile
+The brew file has 4 parts, taps, brew, cask, mas. For the most part `sed` will filter out the parts you need, first dump your `Brewfile`...
+```bash
+brew bundle dump
+```
+...then take a look a how well  `sed` works.
```
Comment (StephenBrown2) [inline gen-config-helper.md:9]: These steps are using `grep`, not `sed`...
```suggestion
...then take a look a how well `grep` works.
```

File: gen-config-helper.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+```bash
+cat Brewfile | grep -e brew | awk '{print "  - "$2 }' | pbcopy
+```
+```bash
+cat Brewfile| grep -e cask | awk '{print "  - "$2 }' | pbcopy
+```
+```bash
+cat Brewfile | grep -e mas |  awk '{print "  - { id: "$4", name: "$2 "}"}' | pbcopy
+```
+
```
Comment (StephenBrown2) [inline gen-config-helper.md:36]: Just a tip, awk can match things and operate on files itself, so you don't need to use grep or cat

```suggestion
You can build up your `config.yml` section by section. I piped the following commands to `pbcopy` and cleaned up in the `config.yml` file directly. 
```bash
awk '/tap/{print "  - "$2}' Brewfile | pbcopy
```
```bash
awk '/brew/{print "  - "$2 }' Brewfile | pbcopy
```
```bash
awk '/cask/{print "  - "$2 }' Brewfile | pbcopy
```
```bash
awk '/mas/{print "  - { id: "$4", name: "$2 "}"}' Brewfile | pbcopy
```

```

File: gen-config-helper.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+cat Brewfile| grep -e tap
+```
+```bash
+cat Brewfile| grep -e brew
+```
+```bash
+cat Brewfile| grep -e cask
+```
+```bash
+cat Brewfile| grep -e mas
```
Comment (StephenBrown2) [inline gen-config-helper.md:20]: grep can operate on files by itself, no need for cat:
```suggestion
grep -e tap Brewfile
```
```bash
grep -e brew Brewfile
```
```bash
grep -e cask Brewfile
```
```bash
grep -e mas Brewfile
```

File: gen-config-helper.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+Depending on how you want to go about this, sometimes I feel clever to pack up a bunch of commands in to an array and execute them, below is one such example without the copying to the clipboard. I personally found that I had too many packages to review so I opted with the above.
+
+```bash
+#!/bin/bash 
+
+backup_commands=(
+  "brew bundle dump"
+  "cat Brewfile | grep -e tap | awk '{print "  - "$2}'"
+  "cat Brewfile | grep -e brew | awk '{print "  - "$2 }'"
+  "cat Brewfile | grep -e mas |  awk '{print "  - { id: "$4", name: "$2 "}"}'"
```
Comment (StephenBrown2) [inline gen-config-helper.md:62]: Application of prior comments:

```suggestion
  "awk '/tap/{print "  - "$2}' Brewfile"
  "awk '/brew/{print "  - "$2 }' Brewfile"
  "awk '/mas/{print "  - { id: "$4", name: "$2 "}"}' Brewfile"
```

File: gen-config-helper.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
+# config.yml helper
+In adapting [mac-dev-playbook](https://github.com/geerlingguy/mac-dev-playbook), preparing the config.yml file can be somewhat tedious. I found the following bash one-liners helpful, you may too. You'll want to clean up the output, but it should get you most of the way there.
+
+### Brewfile
+The brew file has 4 parts, taps, brew, cask, mas. For the most part `sed` will filter out the parts you need, first dump your `Brewfile`...
```
Comment (StephenBrown2) [inline gen-config-helper.md:5]:
```suggestion
The brew file has 4 parts, taps, brew, cask, mas. For the most part `grep` will filter out the parts you need, first dump your `Brewfile`...
```

**Review by StephenBrown2 — CHANGES_REQUESTED** [review]
Looks good. Not sure if it'll get merged, but I left some suggestions that I hope are helpful for you.

File: gen-config-helper.md (hunk 2 above)
Comment (pleasemarkdarkly) [reply gen-config-helper.md:36]: Thank you, these are super helpful and of course I had forgotten awk and grep file manipulation.

**Pull request closed without merging**